### PR TITLE
[exporter/loki] Deprecate the format option

### LINKED ISSUE #12897: [exporter/loki] Deprecate the format option, remove the "body" encoding
I would like to remove the "format" attribute. It's rarely the right thing to do, with the JSON encoding being the recommended one.

Before deprecating and eventually removing it, I would like to hear about use cases from the current users of the Loki Exporter. The current plan is to deprecate this option by v0.57.0, removing it by v0.60.0.

Relates to #12873, but I split this from that as we should be able to do it right now without further refactoring.

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #12897

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: exporter/lokiexporter/config.go [hunk 1]
```diff
@@ -39,8 +39,11 @@ type Config struct {
 	// Labels defines how labels should be applied to log streams sent to Loki.
 	Labels LabelsConfig `mapstructure:"labels"`
 
-	// Allows you to choose the entry format in the exporter
-	Format string `mapstructure:"format"`
+	// Allows you to choose the entry format in the exporter.
+	// Deprecated: [v0.57.0] Only the JSON format will be supported in the future. If you rely on the
+	// "body" format and can't change to JSON, let us know before v0.59.0 by opening a GitHub issue
+	// and we'll work with you to find a solution.	Format string `mapstructure:"format"`
+	Format *string `mapstructure:"format"`
```
Comment (Aneurysm9) [inline exporter/lokiexporter/config.go:46]: Can this be done without an immediate breaking change to an exported struct?  Making this a pointer would enable the different messaging added here for situations where the deprecated value is provided and no value is provided, but I think that with the default config containing a value the exporter will never see a `nil` value as the default will be merged with the user-provided config before being given to the exporter.  Is the differential messaging really required? Can we just warn users who have an effective config referencing the deprecated value?

File: unreleased/loki-deprecate-format.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
```
Comment (Aneurysm9) [inline unreleased/loki-deprecate-format.yaml:2]: This changes the type of an exported field in an exported struct, and is thus a breaking change.

File: exporter/lokiexporter/config.go (hunk 1 above)
Comment (jpkrohling) [reply exporter/lokiexporter/config.go:46]: > but I think that with the default config containing a value the exporter will never see a nil value as the default

That's true, I should have removed it from the default config. 

> Is the differential messaging really required?

I believe so: telling users that they didn't set a value and that the default will change will reduce confusion, as users seeing a generic message based on the effective value would not know where this value is coming from.

File: unreleased/loki-deprecate-format.yaml (hunk 2 above)
Comment (jpkrohling) [reply unreleased/loki-deprecate-format.yaml:2]: That's true. I'll mark it as such.

**Comment by jpkrohling** [comment]
This is ready for review again. Thanks @Aneurysm9!

File: exporter/lokiexporter/config.go [hunk 3]
```diff
@@ -39,8 +39,11 @@ type Config struct {
 	// Labels defines how labels should be applied to log streams sent to Loki.
 	Labels LabelsConfig `mapstructure:"labels"`
 
-	// Allows you to choose the entry format in the exporter
-	Format string `mapstructure:"format"`
+	// Allows you to choose the entry format in the exporter.
+	// Deprecated: [v0.57.0] Only the JSON format will be supported in the future. If you rely on the
+	// "body" format and can't change to JSON, let us know before v0.59.0 by opening a GitHub issue
+	// and we'll work with you to find a solution.	Format string `mapstructure:"format"`
```
Comment (Aneurysm9) [inline exporter/lokiexporter/config.go:45]:
```suggestion
	// and we'll work with you to find a solution.
```

Guessing that wasn't meant to be left there.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**